rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionService.scala

```diff
@@ -178,6 +181,27 @@ class ReportsExecutionService (
       case scala.util.Failure(ex) =>
         logger.error("An error occured when trying to update the cache of last changes", ex)
     }
+
+    future {
+      for {
+        updated <- cachedCompliance.checkAndUpdateCache(updatedNodeIds)
```

Comment:
i'm not sure of why you are doing this:
- need to check if cache is filed for these nodes
- need to invalidate the cache for expired nodes
- but why with a list of nodeid as parameters, and shouldn't it be the job of https://github.com/Normation/rudder/pull/805/files#diff-14ce2ba69947867c1979f20a63aa2d2bL163 ?